File: Capstone Project/Backend side/AssessmentPortal/src/main/java/com/NTeq/AssessmentPortal/Services/ResultService.java
package com.NTeq.AssessmentPortal.Services;

import java.util.List;

import com.NTeq.AssessmentPortal.Dto.ResultDto;
import com.NTeq.AssessmentPortal.Response.SuccessResponse;

/**
 * Service interface for managing result-related operations.
 */
public interface ResultService {
    /**
     * Adds a new result to the system.
     * @param resultDto The DTO containing result details.
     * @return A message indicating the result of the operation.
     */
    SuccessResponse addResult(ResultDto resultDto);
    /**
     * Retrieves a list of all results.
     * @return A list of result DTOs.
     */
    List<ResultDto> getAllResult();
    /**
     * Retrieves a list of all results.
     * @param userEmail   the email of users, whom result should be generated.
     * @return A list of result DTOs.
     */
    List<ResultDto> getresultByEmail(String userEmail);
}

File: Capstone Project/assessment-portal-frontend/src/Services/resultService.js
import axios from 'axios'

const Result_BASE_API_URL = "http://localhost:8080/result";

class resultService{

    saveResult(Result){
        return axios.post(Result_BASE_API_URL + "/addResult" , Result);
    }

    getAll(){
        return axios.get(Result_BASE_API_URL + "/all");
    }

    findByEmailId(email){
        return axios.get(Result_BASE_API_URL + "/" + email);
    }
}

export default new resultService();